### PR TITLE
Documentation for k8s Flannel support

### PR DESCRIPTION
**Pull request by sjberman**
Adding documentation for users to configure Flannel for the k8s-bigip-ctlr. These docs describe
how users can configure their BIG-IP systems and a Kubernetes Node to proeprly support cluster networking in Kubernetes.

Also adding some extended information about how Flannel works with Kubernetes, how the BIG-IP device can be added to the Kubernetes network, and what role the F5 BIG-IP Controller plays in the configuration steps.

### CODE REVIEW

File: docs/kubernetes/flannel-bigip-info.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+Flannel solely used an etcd key-value store to read and write information about Kubernetes Nodes.
+This information included the VTEP (VXLAN tunnel endpoint) of each Node, as well as a subnet within the defined Flannel network range.
+Flannel assigns a different subnet to each Node, and allocates all Pod IP's within these subnets (on their respective Nodes).
+In more recent versions of Kubernetes, Flannel's preferred mode of operation is via the Kubernetes API.
+Etcd is still viable, but is not as common. Flannel provides a `kube-flannel.yml`_ file for use
+as a network addon to kubeadm. This creates a Flannel Pod on each Kubernetes node that runs two containers: install-cni and flanneld.
+Each instance of flanneld now monitors the Kubernetes API for the Nodes that live in the cluster.
+The Nodes will have flannel-specific annotations attached to them, containing similar information to what etcd stores.
+
+Here are the supported annotations:
```
Comment (jputrino) [inline docs/kubernetes/flannel-bigip-info.rst]: @sjberman - This bit about the Annotations is a little confusing. They come from json blobs included in the data section of the ConfigMap, correct? Do they appear as Annotations when you get the Node description?

Comment (sjberman) [reply docs/kubernetes/flannel-bigip-info.rst]: These aren't involved in any ConfigMap. They are simply annotations that are configured as part of the Node description, yes. Does that answer your question?

**Comment by jputrino** [comment]
@sjberman -- Edits are complete; please review for technical accuracy.
[Staged docs](https://s3-us-west-2.amazonaws.com/staging-c2ub89n2qjgt1/f5-ci-docs/pr313/index.html)

File: docs/kubernetes/kctlr-use-bigip-k8s.rst [hunk 2]
```diff
@@ -67,31 +76,63 @@ Create a BIG-IP VXLAN tunnel
...
+
+   .. code-block:: console
+
+      kubectl apply -f kube-flannel.yaml --namespace=kube-system
+
+   \
+
+   .. tip::
+
+      You can also deploy flannel as a `network addon`_ when setting up ``kubeadm``. See :k8sdocs:`Installing a Pod network <setup/independent/create-cluster-kubeadm/#pod-network>` for instructions.
```
Comment (sjberman) [inline docs/kubernetes/kctlr-use-bigip-k8s.rst]: I think the preferred way to deploy the yaml file is via a network addon for kubeadm. We should mention this first, and have an aside that says you could also manually deploy.

File: docs/kubernetes/kctlr-use-bigip-k8s.rst [hunk 3]
```diff
@@ -1,40 +1,49 @@
...
    - BIG-IP v12.1.1
    - Kubernetes v1.7.5
    - Flannel v0.9.0
 
+.. _bigip-k8s-setup:
+
+Add your BIG-IP device to a Kubernetes Cluster using Flannel VXLAN
+==================================================================
+
+This document provides step-by-step instructions for adding a BIG-IP device to a Kubernetes Cluster using Flannel VXLAN. For more information about this integration, see :ref:flannel-bigip-info`.
```
Comment (sjberman) [inline docs/kubernetes/kctlr-use-bigip-k8s.rst]: Missing a backtick here in the ref; link doesn't work.

File: docs/kubernetes/flannel-bigip-info.rst [hunk 4]
```diff
@@ -1,143 +1,116 @@
...
+**For example:**
+
+Node1 has the IP address, MAC address, and Pod IP address shown in the table below.
+
++------------------------------------------------------------------+
+| Kubernetes Node1                                                 |
++==============================================+===================+
+| Node IP address                              | 172.16.2.10       |
++----------------------------------------------+-------------------+
+| MAC address of Nodes flannel VXLAN interface | 98:ba:76:dc:54:fe |
```
Comment (sjberman) [inline docs/kubernetes/flannel-bigip-info.rst]: Node's

File: docs/kubernetes/flannel-bigip-info.rst [hunk 5]
```diff
@@ -1,143 +1,116 @@
...
+At this point, your BIG-IP device knows how to route to the Kubernetes network, but it isn't a part of the network yet. You'll need to set up flannel to be aware of the BIG-IP device. To do so, you'll create an additional flannel subnet and allocate a self IP address from this subnet as the BIG-IP's VTEP.
+
+When setting up a new subnet to use for the BIG-IP, make sure that it doesn't collide with any of the Node subnets.
+
+**For example:**
+Create a new subnet: ``10.244.30.0/24``. Your BIG-IP device's self IP will live within this subnet.
 
-The F5 BIG-IP Controller does most of the heavy lifting in this process, so the user steps are much easier.
-A user simply has to:
+Then, create a subnet for the VXLAN tunnel: ``10.244.30.15/16``. The subnet mask -- ``/16`` -- matches the flannel network range's default subnet mask.
```
Comment (sjberman) [inline docs/kubernetes/flannel-bigip-info.rst]: "create a self-ip", not "create a subnet"

File: docs/kubernetes/flannel-bigip-info.rst [hunk 6]
```diff
@@ -1,143 +1,116 @@
...
-directed at the Kubernetes Pod, will successfully reach its destination!
+Tell flannel about the BIG-IP
+`````````````````````````````
+
+At this point, your BIG-IP device knows how to route to the Kubernetes network, but it isn't a part of the network yet. You'll need to set up flannel to be aware of the BIG-IP device. To do so, you'll create an additional flannel subnet and allocate a self IP address from this subnet as the BIG-IP's VTEP.
+
+When setting up a new subnet to use for the BIG-IP, make sure that it doesn't collide with any of the Node subnets.
+
+**For example:**
+Create a new subnet: ``10.244.30.0/24``. Your BIG-IP device's self IP will live within this subnet.
```
Comment (sjberman) [inline docs/kubernetes/flannel-bigip-info.rst]: I think the word "define" makes more sense than "create" here. The user isn't actually creating anything, but they will have to "come up with" a subnet range that gets put into the Node configuration (podCIDR) at a later step.

Comment (jputrino) [reply docs/kubernetes/flannel-bigip-info.rst]: Gotcha. I was thinking it was like the Host Subnet in OpenShift.

**Comment by jputrino** [comment]
@sjberman - All comments addressed.

**Comment by sjberman** [comment]
@jputrino Looks good to me.

**Pull request merged**